cmd/storage-rest-client.go

```diff
@@ -418,20 +418,35 @@ func (client *storageRESTClient) UpdateMetadata(ctx context.Context, volume, pat
 	return err
 }
 
-func (client *storageRESTClient) DeleteVersion(ctx context.Context, volume, path string, fi FileInfo, forceDelMarker bool) error {
+func (client *storageRESTClient) DeleteVersion(ctx context.Context, volume, path string, fi FileInfo, forceDelMarker bool) StorageDeleteResp {
 	values := make(url.Values)
 	values.Set(storageRESTVolume, volume)
 	values.Set(storageRESTFilePath, path)
 	values.Set(storageRESTForceDelMarker, strconv.FormatBool(forceDelMarker))
 
 	var buffer bytes.Buffer
 	if err := msgp.Encode(&buffer, &fi); err != nil {
-		return err
+		return StorageDeleteResp{Err: err}
 	}
 
 	respBody, err := client.call(ctx, storageRESTMethodDeleteVersion, values, &buffer, -1)
 	defer xhttp.DrainBody(respBody)
-	return err
+	if err != nil {
+		if contextCanceled(ctx) {
+			err = ctx.Err()
+		}
+		return StorageDeleteResp{Err: err}
+	}
+	reader, err := waitForHTTPResponse(respBody)
+	if err != nil {
```

Comment:
You won't need keep alive for this call, this is a quick rename or metadata update. You can simply do gob.Encode() directly.